contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj

```diff
@@ -54,27 +56,30 @@
     :validate [check-valid-dir "Input directory not found"]]
    ["-h" "--help"]])
 
-(defn result->map [{:keys [class prob x-min y-min x-max y-max]}]
-  (hash-map
-   :label class
-   :confidence (int (* 100 prob))
-   :top-left [x-min y-min]
-   :bottom-right [x-max y-max]))
-
-(defn print-results [results]
-  (doseq [_r results]
-    (println (format "Class: %s Confidence=%s Coords=(%s, %s)"
-                     (_r :label)
-                     (_r :confidence)
-                     (_r :top-left)
-                     (_r :bottom-right)))))
-
 (defn process-results [images results output-dir]
-  (dotimes [i (count images)]
-    (let [image (nth images i) _results (map result->map (nth results i))]
-      (println "processing: " image)
-      (print-results _results)
-      (draw/draw-bounds image _results output-dir))))
+  (mapv (fn [image-path result]
```

Comment:
I would use a `run!` here to convey that it is side effectful. It is probably a matter of personal preference though.
```
(defn process-result [output-dir [image-path result]] ...)
```
So that it can be called like so:
```
(run! (partial process-result output-dir) (map vector image-paths results))
```

Reply:
Thanks - I'll refactor it so the side effect is more clear with a `process-result!` and `doall`